docs/best-practice.md

```diff
@@ -0,0 +1,255 @@
+
+### Business Rules
+
+Holds any business rule calculations so they can be reused across different packages e.g. deadline dates that vary by appeal type
+
+### The Common Module
+
+The [Common](/packages/common) contains a series of common functions that are used across
+microservices. The applications use this as an external dependency (`@pins/common`)
+but is included as a local file in the `package.json`.
+
+In Docker Compose, this is included as a mounted volume to `/opt/common`.
+
+In the Dockerfile, this pulls the files from a Docker image called "common" - this
+is actually built separately in the CI/CD pipelines. If you need to build the Docker
+image separately, you must build the common image separately:
+
+```
+docker build -t common ./common
+```
+
+## Dynamic Forms
+
+Dynamic Forms/Journeys are a concept used to make the creation of multi page forms quicker as the service has a large quantity of questions that vary based on appeal type
+
+See:
+
+- [Question components](https://pins-ds.atlassian.net/wiki/spaces/AAPDS/pages/1521877020/Question+components)
+- [Journey Types](/packages/common/src/dynamic-forms)
+- [Dynamic Forms](/packages/forms-web-app/src/dynamic-forms/readme.md)
+
+## Dependencies
+
+You will need to install the dependencies locally, even though we're using
+Docker Compose to run locally.
+
+Run `npm ci` in the root of the repo to install all dependencies.
+
+As we are using a workspace in order to add new dependencies run install in root:
+
+```sh
+npm install <new-package>
+```
+
+Then add a versionless reference for each workspace package that uses it e.g.
+
+```json
+"express": "*"
+```
+
+This ensures the version is handled once across all packages, although we can specify multiple versions if necessary
+
+Docker files omit unrequired packages to keep the image as slim as possible
+
+```sh
+npm ci --workspaces --if-present --omit=dev
+```
+
+## Auth
+
+API clients are added to the express request object by middleware and internally add users' auth tokens when calling APIs `req.appealsApiClient` and `req.docsApiClient`, with no user a client credentials token is sent instead.
+
+Ensure all api requests check for a token, and if necessary checks the user can access that specific resource
+
+## Feature Flags
+
+We have incorporated Azure feature flag functionality into the common package for use across the solution. Feature flags should be set up via terraform and can be configured (enabled/disabled/users set) via portal under the 'app configuration' section, where you can select your desired environment and find all available flags accordingly, flags can be used within the codebase by importing the `isFeatureActive` function from the FeatureFlag file in your current package, i.e;
+
```

Comment:
is it worth reiterating this point here? 
Feature flags - terraform seems to have a bug with syntax when changing user targeting rules for feature flags, change them manually and update the terraform afterwards to avoid this